### ISSUE TITLE
tune.with_resources() not working correctly when being used as an argument of tune.with_parameters()

### ISSUE THREAD

**Issue by wxie2013** [ray-project/ray#30839]
### What happened + What you expected to happen

I'm trying to use all CPUs of a machine for each trial via 

    trainable_with_resources = tune.with_resources(trainable, {"cpu": multiprocessing.cpu_count()}) 

I would expect there will be one trial running at one time. However,  all trials run in parallel when using trainable_with_resources as an argument of tune.with_parameters(), i.e. 

     tune.with_parameters(trainable_with_resources, data = data)

It sounds like the constraints applied by tune.with_resources() is overwritten with default value, i.e. 1 cpu per trial, when using together with tune.with_parameters()

### Versions / Dependencies

Ray.2.1.0

### Reproduction script

    import os, time, ray
    import pandas as pd
    import multiprocessing
    from ray import tune, air
    from hyperopt import hp
    from ray.tune.search.hyperopt import HyperOptSearch
    from ray.tune.search import ConcurrencyLimiter

    # disable dash board. See: https://discuss.ray.io/t/disable-dashboard/8471
    ray.init(include_dashboard=False)

    # 1. Define an objective function.
    def objective(config, data = None):
        time.sleep(10)
        f = open(os.path.join(os.getcwd(), "oo.root"), "w")
        for i in range(10): # 10 epochs
            score = config["a"] ** 2 + config["b"]
            #SCORE 1st apparence which defines the key of the dictionary, i.e. metric="SCORE", 
            # or  return {"SCORE": score}
            tune.report(SCORE=score)  # this or the following return should work
            #return {"SCORE": score}

    # 2. Define a search space.
    search_space = {
        "a": hp.uniform("a", 0, 1),
        "b": hp.uniform("b", 0, 1)
        }

    raw_log_dir = "./ray_log"
    raw_log_name = "example"

    algorithm = HyperOptSearch(search_space, metric="SCORE", mode="max", n_initial_points=1)
    algorithm = ConcurrencyLimiter(algorithm, max_concurrent=8)
    if os.path.exists(os.path.join(raw_log_dir, raw_log_name)) == False:
        print('--- this is the 1st run ----')
    else: #note: restoring described here doesn't work: https://docs.ray.io/en/latest/tune/tutorials/tune-stopping.html 
        print('--- previous run exist, continue the tuning ----')
        algorithm.restore_from_dir(os.path.join(raw_log_dir, raw_log_name))

    # 3. Start a Tune run and print the best result.
    trainable_with_resources = tune.with_resources(objective, {"cpu": multiprocessing.cpu_count()})
    tuner = tune.Tuner(
            tune.with_parameters(trainable_with_resources, data = "aaaa"),
            tune_config = tune.TuneConfig(
                num_samples = 8, # number of tries. too expensive for Brian2
                time_budget_s = 10000, # tot running time in seconds
                search_alg=algorithm,
                ),
            run_config = air.RunConfig(local_dir = raw_log_dir, name = raw_log_name) # where to save the log which will be loaded later
            )

    results = tuner.fit()
    best_result = results.get_best_result(metric="SCORE", mode="max")

### Issue Severity

High: It blocks me from completing my task.

**Comment by matthewdeng** [comment]
Hey @wxie2013, this should be a duplicate of https://github.com/ray-project/ray/issues/29235 which has been fixed as part of https://github.com/ray-project/ray/pull/29740 and will be released as part of Ray 2.2!

Can you try the fix by using a nightly wheel?

**Comment by wxie2013** [comment]
Nice. When will Ray 2.2 be released?

**Comment by matthewdeng** [comment]
Ray 2.2 will be released this month! 

If you are interested in trying an early preview version of Ray 2.2, the generated wheels can directly be found [here](https://buildkite.com/ray-project/oss-ci-build-branch/builds/1383) under the corresponding OS  `Wheels and Jars` jobs, in the `Artifacts` tab. Let me know if you need any help with this!
